### ISSUE TITLE
Implement scheme as local JS form?

### ISSUE THREAD

**Issue by nicebread** [nicebread/RESQUE#4]
Desiderata:

- usability for applicants
- privacy (runs only locally)
- ability to save and load data into the form
- nice to have: export in different formats (JSON, maybe a PDF report for the applicants?)
- low maintenance costs (no server component, easy framework)
- How can hiring committees customize it? --> cf. "menu approach". 
    - e.g. select subsets of indicators
    - e.g. change reward points 
    - But how would that work with xlsx sheets without breaking the accompanying R scripts?

- Applicants can provide up to 12 contributions - either Publications, Data sets, or Software. We need these 12 (<- customizable) slots, and that they can change between the slots. (Maybe tabs?)
- Enter author metadata (last name, maybe more items later) only once
- Include version number in json-definition and in saved json --> Warning if version number does not match
- For transparency, applicants should be able to see their reward points. Hence, implement scoring logic into website.
   - Put scores into definition json

Start with the [RESQUE-Pubs scheme](https://docs.google.com/spreadsheets/d/1SUhwiZnhFCjy5COga3utGO15KT2wD4Pk/edit#gid=1896772187) - this is mostly complete.

**Comment by nicebread** [comment]
Idee: Indikatoren mit logischen Abhängigkeit kommen in ein Subset, das (im menu approach) nur als ganzes an- oder abgewählt werden kann.

**Comment by nicebread** [comment]
Do we need a minimal validator of the definition-json?

I imagine the scenario, that a committee removes a set of items (which they don't want), but thereby break the logic of points, as another item refers to a deleted item in a condition.

**Comment by nicebread** [comment]
TODOs:

- [ ] add "eval_dimension" to the json (defaults to "", but can be provided for each indicator). (Can be "Rigor", "Impact", maybe more stuff)

**Comment by alpkaanaksu** [comment]
> Add export option of all fields, so that committees get an overview of all questions

Just add `?mode=preview` to the end of the URL. (i.e. [nicebread.github.io/RESQUE/web?mode=preview](nicebread.github.io/RESQUE/web?mode=preview))

**Comment by alpkaanaksu** [comment]
> Include version number in json-definition and in saved json --> Warning if version number does not match
<img width="402" alt="Screenshot 2023-08-10 at 00 50 41" src="https://github.com/nicebread/RESQUE/assets/68744864/4bd8fdf5-4fdb-4e38-a675-67aecadf592e">

Maybe show only a warning instead of an error and load the data regardless?

**Comment by alpkaanaksu** [comment]
Add scoring information to JSON and implement scoring logic: 8e48d1b2b47dbd8c080b9f080e2489d1e7fb4370

<img width="462" alt="Screenshot 2023-08-11 at 20 46 24" src="https://github.com/nicebread/RESQUE/assets/68744864/420f993b-028b-40fa-9f79-3b704a0ae2b6">

**Comment by alpkaanaksu** [comment]
> Do we need a minimal validation of the definition-json?

> I imagine the scenario, that a committee removes a set of items (which they don't want), but thereby break the logic of points, as another item refers to a deleted item in a condition.

The `pick` functions should handle this.

**Comment by alpkaanaksu** [comment]
> add "eval_dimension" to the json (defaults to "", but can be provided for each indicator). (Can be "Rigor", "Impact", maybe more stuff)

Implemented in 7ac49bafb30a0fd0eae70b2321cb3a3117553d28.

<img width="457" alt="Screenshot 2023-08-12 at 13 11 47" src="https://github.com/nicebread/RESQUE/assets/68744864/a7023aea-1db6-4c69-bf00-811d7bf322c9">

Nothing is shown if the "eval_dimension" field is left out.

**Comment by alpkaanaksu** [comment]
Everything mentioned in this huge issue is now implemented (except validation, I created a new issue for that).
